src/components/widgets/Form.js

```diff
@@ -0,0 +1,315 @@
+import React, { Component } from 'react'
+import { RadioGroup, Radio } from 'react-radio-group'
+import Button from '../Button.js'
+import "../../styles/form.css"
+
+export default class Form extends Component {
+  constructor( props ) {
+    super( props )
+    this.state = {}
+    this.setFieldState = this.setFieldState.bind( this )
+  }
+
+  componentWillMount() {
+    this.setupState()
+  }
+
+  setupState() {
+    const { sections } = this.props
+    sections.forEach( section => {
+      section.fields.forEach ( field => {
+        const { name, defaultValue } = field
+        this.setState({[ name ]: defaultValue })
+        if ( field.type === "nested-dropdown" ) {
+          const nestedDropdownValue = field.options.filter( option => {
+            return option.value === defaultValue
+          })[0].field
+          this.setState({
+            [ nestedDropdownValue.name ]: nestedDropdownValue.defaultValue
+          })
+        }
+      })
+    })
+  }
+
+  setFieldState( state, stateValue ) {
+    this.setState({
+      [ state ]: stateValue
+    })
+  }
+
+  submitWidget( state ) {
+    const widgetData = {}
+    for ( let key in state ) {
+      widgetData[key] = state[key]
+      widgetData.y = '10px'
+      widgetData.x = '20px'
+    }
+    console.log('You submitted a widget with this data:', widgetData)
+  }
+
+  render() {
+    const { type, sections, description, image } = this.props
+
+    return (
+      <div className="widget-form-container">
+        <div className="widget-form-header">
+          <h1>{type}</h1>
+        </div>
+        <form
+          onSubmit={() => this.submitWidget( this.state )}
```

Comment:
Since this function is executed within this context, it has access to `this.state`.  We can remove the parameter, and just allow the function to access `this.state`.  This also means that we should be able to change this to
```jsx
onSubmit={this.submitWidget.bind( this )}
```